### PR TITLE
Fix lock clock

### PR DESCRIPTION
**Pull request by scooterboo**
## Description
Ignores ClientLockGameTimeNotify so that lock time works correctly. Polls PROP_IS_GAME_TIME_LOCKED at login to find out if you are in the middle of a quest with time locking. Much thanks to @LloydTheGreenNinjagc and @Hartie95 

Ya, I can't make heads or tails of what ClientLockGameTimeNotify does except it always is sending a false.

## Type of changes

- [x] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/java/emu/grasscutter/game/world/World.java [hunk 1]
```diff
@@ -56,6 +57,7 @@ public World(Player player, boolean isMultiplayer) {
         this.entity = new EntityWorld(this);
         this.worldLevel = player.getWorldLevel();
         this.isMultiplayer = isMultiplayer;
+        timeLocked = player.getProperty(PlayerProperty.PROP_IS_GAME_TIME_LOCKED) != 0;
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/game/world/World.java]:
```suggestion
        this.timeLocked = player.getProperty(PlayerProperty.PROP_IS_GAME_TIME_LOCKED) != 0;
```

File: src/main/java/emu/grasscutter/server/packet/recv/HandlerClientLockGameTimeNotify.java [hunk 2]
```diff
@@ -11,6 +12,8 @@ public final class HandlerClientLockGameTimeNotify extends PacketHandler {
     @Override
     public void handle(GameSession session, byte[] header, byte[] payload) throws Exception {
         var packet = ClientLockGameTimeNotify.parseFrom(payload);
-        session.getPlayer().getWorld().lockTime(packet.getIsLock());
+//        session.getPlayer().getWorld().lockTime(packet.getIsLock());
+        //TODO: figure out what to implement here
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/server/packet/recv/HandlerClientLockGameTimeNotify.java]:
```suggestion
        // session.getPlayer().getWorld().lockTime(packet.getIsLock());
        // TODO: figure out what to implement here
```

File: src/main/java/emu/grasscutter/server/packet/recv/HandlerClientLockGameTimeNotify.java [hunk 3]
```diff
@@ -11,6 +12,8 @@ public final class HandlerClientLockGameTimeNotify extends PacketHandler {
...
+        if (packet.getIsLock()) Grasscutter.getLogger().warn("Tell Nazrin to PANIK because ClientLockGameTimeNotify is TRUE!");
```
Comment (KingRainbow44) [inline src/main/java/emu/grasscutter/server/packet/recv/HandlerClientLockGameTimeNotify.java]:
```suggestion
        if (packet.getIsLock()) Grasscutter.getLogger().warn("Invalid 'ClientLockGameTimeNotify' received; value is true. (please report to development channel)");
```
consider changing to something more reasonable, example shown above

**Review by KingRainbow44 — CHANGES_REQUESTED** [review]

**Review by scooterboo — COMMENTED** [review]
Yep

**Review by scooterboo — COMMENTED** [review]
test

**Review by KingRainbow44 — APPROVED** [review]

**Pull request merged**